code/backend/batch/utilities/tools/QuestionAnswerTool.py

```diff
@@ -1,182 +1,113 @@
+        retrieved_documents = []
+        source_documents = []
+        i = 1
+        for page in search_results.by_page():
+            for document in page:
+                retrieved_documents.append(
+                    {f"[doc{i}]": {"content": document.get("content")}}
                 )
+                source_documents.append(
+                    SourceDocument(
+                        id=document.get("id"),
+                        content=document.get("content"),
+                        title=document.get("title"),
+                        source=document.get("source"),
+                        chunk=document.get("chunk"),
+                        offset=document.get("offset"),
+                        page_number=document.get("page_number"),
+                    )
+                )
+                i += 1
 
-        example_prompt = ChatPromptTemplate.from_messages(
-            [
-                HumanMessagePromptTemplate.from_template(
-                    self.config.prompts.answering_user_prompt
-                ),
-                AIMessagePromptTemplate.from_template("{answer}"),
-            ]
-        )
-
-        few_shot_prompt = FewShotChatMessagePromptTemplate(
-            example_prompt=example_prompt,
-            examples=examples,
+        ret_docs_json = json.dumps(
+            {"retrieved_documents": retrieved_documents}, separators=(",", ":")
         )
+        question_with_sources = self.config.prompts.answering_user_prompt.replace(
+            "{sources}", ret_docs_json
+        ).replace("{question}", question)
 
-        answering_prompt = ChatPromptTemplate.from_messages(
-            [
-                SystemMessage(content=self.config.prompts.answering_system_prompt),
-                few_shot_prompt,
-                SystemMessage(content=self.env_helper.AZURE_OPENAI_SYSTEM_MESSAGE),
-                MessagesPlaceholder("chat_history"),
-                HumanMessagePromptTemplate.from_template(
-                    self.config.prompts.answering_user_prompt
-                ),
+        if self.config.prompts.use_on_your_data_format:
+            example_question = self.config.prompts.answering_user_prompt.replace(
+                "{sources}", self.config.example.documents
+            ).replace("{question}", self.config.example.user_question)
+            messages = [
+                # {"role": "system", "content": f"{self.config.prompts.answering_system_prompt} {self.env_helper.AZURE_OPENAI_SYSTEM_MESSAGE}"}, # TODO is this second system message needed?
+                {
+                    "role": "system",
+                    "content": self.config.prompts.answering_system_prompt,
+                },
+                {"role": "system", "name": "example_user", "content": example_question},
+                {
+                    "role": "system",
+                    "name": "example_assistant",
+                    "content": self.config.example.answer,
+                },
```

Comment:
I don't think we need `name` here, and also the roles should be `system`, then `user`, then `assistant`